im-sdk-core/src/main/java/com/easemob/im/server/api/loadbalance/FixedEndpointProvider.java

```diff
@@ -0,0 +1,66 @@
+package com.easemob.im.server.api.loadbalance;
+
+import com.easemob.im.server.EMProperties;
+import com.easemob.im.server.exception.EMInvalidArgumentException;
+import reactor.core.publisher.Mono;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class FixedEndpointProvider implements EndpointProvider {
+
+    private static final Pattern BASE_URI = Pattern.compile("^(https|http)://([a-z0-9.:]{0,64})$");
+
+    private final EMProperties properties;
+
+    public FixedEndpointProvider(EMProperties properties) {
+        this.properties = properties;
+    }
+
+    @Override
+    public Mono<List<Endpoint>> endpoints() {
+        String baseUri = this.properties.getBaseUri();
+        validateBaserUri(baseUri);
+
+        if (search(baseUri, ":") > 1) {
+            String[] split = baseUri.split(":");
+
+        }
+
+        String[] splitBaseUri = baseUri.split("://");
```

Comment:
There's a URI class which can parse an URI for you, no need to re-invent it here.